### PR TITLE
Reuse e2e-kind script in Github Action

### PR DESCRIPTION
**Pull request by markusthoemmes**
As per title.

### CODE REVIEW

**Comment by markusthoemmes** [comment]
/hold

not quite there yet

File: test/e2e-kind.sh [hunk 1]
```diff
@@ -20,33 +20,35 @@ set -euo pipefail
 
 KOURIER_GATEWAY_NAMESPACE=kourier-system
 KOURIER_CONTROL_NAMESPACE=knative-serving
+CLUSTER_SUFFIX=${CLUSTER_SUFFIX:-cluster.local}
 
 export KO_DOCKER_REPO=kind.local
```
Comment (nak3) [inline test/e2e-kind.sh:25]: nit:
`KO_DOCKER_REPO` is defined in `.github/workflows/kind-e2e.yaml`. Should we remove this line or define it only when it isn not defined like `CLUSTER_SUFFIX`?

File: test/e2e-upgrade-kind.sh [hunk 2]
```diff
@@ -23,7 +23,6 @@ KOURIER_CONTROL_NAMESPACE=knative-serving
 TEST_NAMESPACE=serving-tests
 
 export KO_DOCKER_REPO=kind.local
```
Comment (nak3) [inline test/e2e-upgrade-kind.sh:25]: Same with `e2e-kind.sh`. 
Additionally, `KO_DOCKER_REPO` is defined twice in this script L#25 and L#59.

**Review by nak3 — COMMENTED** [review]
LGTM except for a nit.

**Review by nak3 — APPROVED** [review]
/lgtm

**Pull request merged**